peps/pep-0792.rst

```diff
@@ -0,0 +1,371 @@
+     quarantined.
+
+   * Project maintainers can "archive" a project. Archiving a project
+     disables new release and distribution uploads to that project,
+     but otherwise has no effect on the project's downloadability.
+
+   Project statuses are machine-readable *in principle*, but are not currently
+   exposed via any of PyPI's APIs. Instead, PyPI renders project statuses on
+   each project's user-facing (i.e. non-index) HTML.
+
+In summary: there are multiple ways to communicate the "status" of a project in
+Python packaging, but none of these mechanisms are *simultaneously* (1)
+machine-readable, (2) general, (3) index-agnostic, and (4) updatable across
+the entire project (instead of per-release or per-distribution).
+
+.. csv-table::
+    :header: "Mechanism", "Machine-readable?", "General?", "Index-agnostic?", "Project updatable?"
+
+    "Trove classifiers", "✅", "✅", "✅", "❌"
+    "Yanking", "✅", "❌", "✅", "✅"
+    "Project statuses", "✅", "✅", "❌", "✅"
+
+This PEP proposes adopting PyPI's project statuses as an index-agnostic
+mechanism, satisfying all four conditions.
+
+Specification
+=============
+
+This PEP specifies two aspects: a set of project status markers,
+as well as their presentation in the standard HTML and JSON indices.
+
+Project status markers
+----------------------
+
+This PEP proposes the following project status markers.
+
+A project **MUST** have exactly one project status marker at any given time.
+
+Indices **MAY** implement any subset of the status markers specified in this
+PEP, as applicable.
+
+``active``
+~~~~~~~~~~
+
+Description: The project is active. This is the default status for a project.
+
+Index semantics: none.
+
+Installer semantics: none.
+
+``archived``
+~~~~~~~~~~~~
+
+Description: The project does not expect to be updated in the future.
+
+Index semantics:
+
+* The index hosting the project **MUST NOT** allow uploads of new distributions to
+  the project.
+* The index **MUST** offer existing distributions of the project for download.
+
+Installer semantics:
+
+* Installers **MAY** produce warnings about a project's archival.
+
+``quarantined``
+~~~~~~~~~~~~~~~
+
+Description: The project is considered generally unsafe for use, e.g. due to
+malware.
+
+Index semantics:
+
+* The index hosting the project **MUST NOT** allow uploads of new distributions to
+  the project.
+* The index **MUST NOT** offer any distributions of the project for download.
+
+Installer semantics:
+
+* Installers **MAY** produce warnings about a project's quarantine, although
+  doing so is effectively moot (as the index will not offer any distributions
+  for installation).
+
+``deprecated``
+~~~~~~~~~~~~~~
+
+Description: The project is considered obsolete, and may have been superseded
+by another project.
+
+Index semantics:
+
+* The index hosting the project **MUST** allow uploads of new distributions to
+  the project.
+* The index **MUST** offer existing distributions of the project for download.
```

Comment:
This is the same as 'active' (/status quo), which has semantics: none. Perhaps either list these explicit semantics for 'active', or say that there are no special semantics for an index here?

Reply:
Thanks, deduping by saying "same as active" makes sense to me here!